opslevel/resource_opslevel_integration_aws.go

```diff
@@ -188,18 +189,21 @@ func (r *IntegrationAwsResource) Update(ctx context.Context, req resource.Update
 		resp.Diagnostics.Append(diags...)
 		return
 	}
-	regionOverride, diags := ListValueToStringSlice(ctx, planModel.RegionOverride)
-	if diags != nil && diags.HasError() {
-		resp.Diagnostics.Append(diags...)
-		return
-	}
 	input := opslevel.AWSIntegrationInput{
 		Name:                 opslevel.RefOf(planModel.Name.ValueString()),
 		IAMRole:              opslevel.RefOf(planModel.IamRole.ValueString()),
 		ExternalID:           opslevel.RefOf(planModel.ExternalID.ValueString()),
 		OwnershipTagOverride: opslevel.RefOf(planModel.OwnershipTagOverrides.ValueBool()),
 		OwnershipTagKeys:     ownershipTagKeys,
-		RegionOverride:       &regionOverride,
+	}
+
+	if !planModel.RegionOverride.IsNull() || !stateModel.RegionOverride.IsNull() {
```

Comment:
Does this properly handle if the regions are passed in from the 

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/region
or
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/regions

?

Reply:
Yes no worries here. I tested this out and the `region` data source returns values that are known to the plan after the read operation and before the update operation begins. So IsUnknown() doesn't apply during the Update operation